node/node_syncing.go

```diff
@@ -410,6 +410,8 @@ func (node *Node) CalculateResponse(request *downloader_pb.DownloaderRequest, in
 
 	// this is the out of sync node acts as grpc server side
 	case downloader_pb.DownloaderRequest_NEWBLOCK:
+		node.stateMutex.Lock()
```

Comment:
is it a new mutex? do we have the race condition?

Reply:
Open race condition